support-frontend/app/services/ProductCatalogService.scala

```diff
@@ -0,0 +1,58 @@
+package services
+
+import com.gu.okhttp.RequestRunners.FutureHttpClient
+import com.gu.rest.WebServiceHelper
+import io.circe.{Decoder, Json, JsonObject}
+import io.circe.generic.semiauto.deriveDecoder
+import org.apache.pekko.actor.ActorSystem
+
+import java.util.concurrent.atomic.AtomicReference
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+
+case class ProductCatalogServiceError(message: String) extends Throwable
+object ProductCatalogServiceError {
+  implicit val decoder: Decoder[ProductCatalogServiceError] = deriveDecoder
+}
+
+trait ProductCatalogService extends WebServiceHelper[ProductCatalogServiceError] {
+  val client: FutureHttpClient
+  override val wsUrl: String
+  override val httpClient: FutureHttpClient = client
+
+  def get(): Future[JsonObject] = {
+    get[JsonObject](endpoint = "product-catalog.json")
+  }
+}
+
+class ProdProductCatalogService(val client: FutureHttpClient) extends ProductCatalogService {
+  val wsUrl: String = "https://product-catalog.guardianapis.com"
+}
+
+class CodeProductCatalogService(val client: FutureHttpClient) extends ProductCatalogService {
+  val wsUrl: String = "https://product-catalog.code.dev-guardianapis.com"
+}
+
+class CachedProductCatalogService(system: ActorSystem, productCatalogService: ProductCatalogService)(implicit
+    ec: ExecutionContext,
+) {
+  private val json = new AtomicReference[JsonObject](JsonObject())
+  private def update() = {
+    productCatalogService.get().map(json.set)
+  }
+  def get(): JsonObject = json.get()
+
+  system.scheduler.scheduleWithFixedDelay(0.minutes, 10.minutes) { () =>
+    {
+      update()
+    }
+  }
+}
+
+class CachedProductCatalogServiceProvider(
+    codeCachedProductCatalogService: CachedProductCatalogService,
+    prodCachedProductCatalogService: CachedProductCatalogService,
+) {
+  def forUser(isTestUser: Boolean) =
+    if (isTestUser) codeCachedProductCatalogService else prodCachedProductCatalogService
+}
```

Comment:
I've not extended the `TouchPointService` here, as I was creating more bootstrap code than this implementation is which is often a sign of the wrong abstraction being used.

Happy to move this over if there is benefit to that.

Reply:
could also be a sign that the touchpointservice could be simplified!  I think one of the strengths of this code base is its consistency (as opposed to support-service-lambdas for example)

Reply:
Happy to try and simplify outside of this PR - do you think I should go ahead and bootstrap those abstractions then?

Reply:
happy to leave the decision up to you as you've seen the extra boilerplate, if you're happy as is then leave it if it's working

Reply:
I might have a go at creating a configless Touchpoint service as a refactor based on this.